### PR TITLE
Merge API documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added information about the `merge` API in the docs

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/api/http-api.mdx [hunk 1]
```diff
@@ -531,7 +531,57 @@ http -a <source_write_key>: <DATA_PLANE_URL>/v1/alias < alias.json
 | `traits` | Object | Optional | Dictionary of the properties or traits associated with the group, such as `email` or `name`. |
 | `timestamp` | DateTime| Optional | The timestamp of the message's arrival. If you are passing the timestamp in the event, make sure it conforms to the ISO 8601 date format `yyyy-MM-ddTHH:mm:ss.SSSZ`. For example: `2022-02-01T19:14:18.381Z` |
 
-## 11. Batch
+## 11. Merge
```
Comment (ameypv-rudder) [inline docs/api/http-api.mdx:534]: @achettyiitr please confirm if we should include `merge` as a part of the HTTP API spec.

File: docs/event-spec/standard-events/merge.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+In the above example, `type` refers to a unique user identifier like email, phone number, device ID, etc. that can be associated to a given customer profile.
+
+## Merge fields
+
+Apart from the <Link to="/event-spec/standard-events/common-fields/">Common fields</Link>, the `merge` call accepts the following fields:
+
+| **Field**    | **Type** | **Presence** | **Description**      |
+| :----------- | :------- | :----------- | :--------------------------------------------------------------------------------------------------------------- |
+| `userId`      | String   | Required, if `anonymousId` is not present.     | The unique user identifier.                             |
+| `anonymousId` | String | Required, if `userId` is not present. | The anonymous ID associated with a user or visitor. |
```
Comment (ameypv-rudder) [inline docs/event-spec/standard-events/merge.mdx:45]: @achettyiitr should these fields be required in the event? If both are present, does RudderStack auto-merge them as we have mentioned here: https://www.rudderstack.com/docs/features/identity-resolution/#merging-identifiers-with-rudderstack:~:text=to%20the%20warehouse.-,RudderStack%20auto%2Dmerges,-the%20following%20user

File: docs/event-spec/standard-events/merge.mdx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+| `mergeProperties` | Object   | Required     | Includes the user properties to be merged and connected to a given user profile. |
+
+## FAQ
```
Comment (ameypv-rudder) [inline docs/event-spec/standard-events/merge.mdx:48]: @achettyiitr Should we also add a FAQ on the difference between `alias` and `merge` APIs?

File: docs/event-spec/standard-events/merge.mdx (hunk 2 above)
Comment (achettyiitr) [reply docs/event-spec/standard-events/merge.mdx:45]: Yes. Any of userId / anonymousId is required. Otherwise, Gateway will throw the error.

**Review by achettyiitr — APPROVED** [review]
LGTM

File: docs/api/http-api.mdx (hunk 1 above)
Comment (achettyiitr) [reply docs/api/http-api.mdx:522]: Yes. Let's add them.

Comment (achettyiitr) [reply docs/api/http-api.mdx:522]: Also, mention the corresponding blocks for ID resolution as well.

**Review by ishachopra0401 — APPROVED** [review]

**Comment by ishachopra0401** [comment]
sidebar and postnav files need updation.

**Comment by ameypv-rudder** [comment]
> sidebar and postnav files need updation.

yep. purposefully didn't update then as would've had to fix 100 merge conflicts otherwise :p

**Pull request merged**